postgre-client/src/tasks.rs

```diff
@@ -93,51 +97,80 @@ impl PgClient {
         }
     }
 
-    pub async fn update_tasks_and_attempts(
-        &self,
-        data: Vec<UpdatedTask>,
-    ) -> Result<(), IndexDbError> {
-        let mut query_builder: QueryBuilder<'_, Postgres> =
-            QueryBuilder::new("UPDATE tasks SET tsk_status = tmp.tsk_status, tsk_attempts = tsk_attempts+1, tsk_error = tmp.tsk_error FROM (");
+    pub async fn update_tasks(&self, data: Vec<UpdatedTask>) -> Result<(), IndexDbError> {
+        let mut query_builder: QueryBuilder<'_, Postgres> = QueryBuilder::new(
+            "
+            UPDATE tasks 
+            SET 
+                task_status = tmp.task_status, 
+                etag = tmp.etag, 
+                last_modified_at = tmp.last_modified_at, 
+                mutability = tmp.mutability, 
+                next_refresh_at = NOW + INTERVAL '1 day' 
+                FROM (
+        ",
+        );
 
-        query_builder.push_values(data, |mut b, key| {
-            let tsk = UrlWithStatus::new(key.metadata_url.as_str(), false); // status is ignoring here
-            b.push_bind(tsk.metadata_url);
-            b.push_bind(key.status);
-            b.push_bind(key.error);
+        query_builder.push_values(data, |mut b, task| {
+            let url = UrlWithStatus::new(task.metadata_url.as_str(), false); // status is ignored here
+            b.push_bind(url.metadata_url);
+            b.push_bind(task.status);
+            b.push_bind(task.etag.clone());
+            b.push_bind(task.last_modified_at);
+            b.push_bind(task.mutability);
         });
 
-        query_builder.push(") as tmp (tsk_metadata_url, tsk_status, tsk_error) WHERE tasks.tsk_metadata_url = tmp.tsk_metadata_url;");
+        query_builder.push(") as tmp (task_status, etag, last_modified_at, mutability) WHERE tasks.metadata_hash = tmp.metadata_hash;");
 
         let query = query_builder.build();
         query.execute(&self.pool).await?;
 
         Ok(())
     }
 
-    pub async fn get_pending_tasks(
+    pub async fn update_tasks_attempt_time(&self, data: Vec<String>) -> Result<(), IndexDbError> {
```

Comment:
Probably as a next ticket would be to make it more flexible, like `data:Vec<(String,Duration)>` to consider the cache validity, or any other logic we want to put for the next retry

Reply:
MTG-1495